### DISCUSSION TITLE
How to make the static site generator better?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#16] [Ask a question] [answered]
If you are using Yellow to generate static pages, let me know how it works for you. 

Here are two things on my list that could be improved:
- better build/clean interaction: Add support for current directory, also make this fail-save to use. For example to clean up generated thumbnails.
- better build errors: There's a case when target directory is read-only and all writes attempts fail. The command line shows that the build failed, but not why.

**Comment by paulsullivan** [comment]
I have a problem with building static blog pages.

ERROR building content location '/blog/author:yellow/', HTTP/1.1 500 Server error:
Can't write file 'latest/blog/author:yellow/index.html'!

The problem seems to be the colon in the url.  Everything else builds fine.  Using latest Wamp on Windows 7.  I may just alter the blog template to use a "author/yellow" format.

**Comment by markseuffert** [comment]
Good to know! Could be a Windows problem, it doesn't happen on Linux and Mac file systems.

**Comment by paulsullivan** [comment, marked as answer]
The colon ":" is an illegal character for windows file names so that is almost certainly the problem.  It might be worth having a look at changing this format as I'm not going to be the only person trying to use Wamp or Xamp.

**Comment by markseuffert** [comment]
Then there's no easy fix. It would need a bigger change, just for Windows, what do others think about it?

**Comment by markseuffert** [comment]
Should be fixed, see also #125
